### PR TITLE
remove api route /network/total-staked

### PR DESCRIPTION
**Pull request by miiu96**
- Remove API route `/network/total-staked`
- Added the values returned by /total-staked API in /network/economics
- Added total topUp value in /network/economics

### CODE REVIEW

File: node/stakeValuesProc/stakeValuesProcessor.go [hunk 1]
```diff
@@ -1,102 +1,97 @@
...
 
-	return tsp.totalStakedValue, nil
+	return &api.StakeValues{
+		TotalStaked: totalStaked,
+		TopUp:       topUp,
+	}, nil
 }
 
-func (tsp *totalStakedValueProcessor) updateTotalStakedValue() error {
+func (tsp *totalStakedValueProcessor) calculateStakedValueAndTopUp() (*big.Int, *big.Int, error) {
```
Comment (iulianpascalau) [inline node/stakeValuesProc/stakeValuesProcessor.go:66]: computeStakedValueAndTopUp *

Comment (miiu96) [reply node/stakeValuesProc/stakeValuesProcessor.go]: done.

File: data/api/apiBlock.go [hunk 2]
```diff
@@ -40,3 +41,9 @@ type MiniBlock struct {
 	DestinationShard uint32                              `json:"destinationShard"`
 	Transactions     []*transaction.ApiTransactionResult `json:"transactions,omitempty"`
 }
+
+// StakeValues is structure that contains totalStakedValue and topUpValue
```
Comment (bogdan-rosianu) [inline data/api/apiBlock.go:45]: is the *

File: node/stakeValuesProc/stakeValuesProcessor.go [hunk 3]
```diff
@@ -1,102 +1,97 @@
-package totalStakedAPI
+package stakeValuesProc
```
Comment (bogdan-rosianu) [inline node/stakeValuesProc/stakeValuesProcessor.go:1]: `stakevalueprocessor` 
- no camel case in the package name
- no abbreviations

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: node/stakeValuesProc/stakeValuesProcessor.go (hunk 3 above)
Comment (miiu96) [reply node/stakeValuesProc/stakeValuesProcessor.go:1]: done.

File: data/api/apiBlock.go (hunk 2 above)
Comment (miiu96) [reply data/api/apiBlock.go:45]: done.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

**Review by AdoAdoAdo — DISMISSED** [review]

File: node/stakeValuesProcessor/disabledStakeValuesProcessor.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package stakeValuesProcessor
```
Comment (iulianpascalau) [inline node/stakeValuesProcessor/disabledStakeValuesProcessor.go:1]: you can get rid of the code cov issues if you place this file in ./disabled package

File: node/stakeValuesProc/stakeValuesProcessor.go [hunk 5]
```diff
@@ -1,102 +1,97 @@
...
-	if time.Since(tsp.lastComputeTime) < tsp.cacheDuration {
-		tsp.mutex.RLock()
-		defer tsp.mutex.RUnlock()
-
-		return tsp.totalStakedValue, nil
-	}
-
+func (tsp *totalStakedValueProcessor) GetTotalStakedValue() (*api.StakeValues, error) {
 	tsp.mutex.Lock()
 	defer tsp.mutex.Unlock()
```
[the remark this answers is not in the document]
Comment (miiu96) [reply node/stakeValuesProc/stakeValuesProcessor.go]: done.

File: node/stakeValuesProcessor/disabledStakeValuesProcessor.go (hunk 4 above)
Comment (miiu96) [reply node/stakeValuesProcessor/disabledStakeValuesProcessor.go]: done.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by stefanandy — APPROVED** [review]
System Test Passed

**Pull request merged**